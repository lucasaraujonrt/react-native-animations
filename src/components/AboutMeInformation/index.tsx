import React from 'react';
import { View } from 'react-native';

// import { Container } from './styles';

interface IAboutMeInformationProps {
  currentIndex: number;
}

// todo get the current Index and return the information animated

const AboutMeInformation = ({ currentIndex }: IAboutMeInformationProps) => (
  <View key={currentIndex} />
);

export default AboutMeInformation;
