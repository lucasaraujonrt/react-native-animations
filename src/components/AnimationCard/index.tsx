import React from 'react';

import navigationService from '@mobile/services/navigation';

import * as S from './styles';

interface IAnimationCardProps {
  name: string;
  type: 'moti' | 'reanimated' | 'native';
  navigateTo: string;
  icon: React.ReactNode;
  disabled: boolean;
}

const AnimationCard = ({
  navigateTo,
  type,
  name,
  icon,
  disabled,
}: IAnimationCardProps) => (
  <S.Container
    {...{ onPress: () => navigationService.navigate(navigateTo), disabled }}
  >
    <S.Wrapper>
      <S.WrapperIcon>{icon}</S.WrapperIcon>
      <S.WrapperText>
        <S.Text>{name}</S.Text>
      </S.WrapperText>
    </S.Wrapper>
    <S.SecondWrapper>
      <S.WrapperType>
        <S.Type>{type}</S.Type>
      </S.WrapperType>
    </S.SecondWrapper>
  </S.Container>
);

export default AnimationCard;
