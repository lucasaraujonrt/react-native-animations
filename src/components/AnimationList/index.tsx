import { MotiView } from 'moti';
import React from 'react';
import { View } from 'react-native';
import { Easing } from 'react-native-reanimated';

import { getDirection } from '@mobile/animation/helpers/directions';

// import { Container } from './styles';

interface IAnimationListProps {
  index: number;
  children: React.ReactElement;
}

const AnimationList = ({ children, index }: IAnimationListProps) => (
  <MotiView
    delay={300 * index}
    from={{
      opacity: 0,
      scale: 0.5,
      [getDirection(index)]: -20,
    }}
    animate={{
      opacity: 1,
      scale: 1,
      [getDirection(index)]: 0,
    }}
    transition={{
      type: 'spring',
      duration: 3500,
      scale: {
        delay: 50,
      },
      easing: Easing.out(Easing.bounce),
    }}
  >
    {children}
  </MotiView>
);

export default AnimationList;
