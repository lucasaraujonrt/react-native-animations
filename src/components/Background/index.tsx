import React from 'react';
import { StatusBar } from 'react-native';

import * as S from './Background.styles';

interface IBackground {
  children: React.ReactNode;
  backgroundColor?: string;
  scrollView?: boolean;
  barStyle: 'light-content' | 'default' | 'dark-content';
}

const Background = ({
  backgroundColor,
  children,
  scrollView,
  barStyle,
}: IBackground) => (
  <>
    <StatusBar
      barStyle={barStyle}
      animated
      translucent
      backgroundColor={backgroundColor}
    />
    {scrollView ? (
      <S.ScrollView backgroundColor={backgroundColor}>
        {children}
      </S.ScrollView>
      ) : (
        <S.Container backgroundColor={backgroundColor}>{children}</S.Container>
      )}
  </>
  );

export default Background;
