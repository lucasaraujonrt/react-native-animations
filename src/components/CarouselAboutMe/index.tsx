import { MotiView } from 'moti';
import React from 'react';

import navigationService from '@mobile/services/navigation';

import * as S from './styles';

interface Data {
  id: number;
  title: string;
  icon: React.ReactNode;
  disabled: boolean;
  navigateTo?: string;
}

interface ICarouselAboutMeProps {
  data: Data[];
}

const CarouselAboutMe = ({ data }: ICarouselAboutMeProps) => (
  <S.Map>
    {data.map((item, index) => (
      <MotiView
        delay={(1200 * index) / 2}
        from={{
          opacity: 0,
          translateY: -20,
        }}
        animate={{
          opacity: 1,
          translateY: 0,
        }}
      >
        <S.Container
          {...{
            disabled: item.disabled,
            key: `key-${index * item.id}`,
            onPress: () =>
              item.navigateTo && navigationService.navigate(item.navigateTo),
          }}
        >
          <S.WrapperIcon>{item.icon}</S.WrapperIcon>
          <S.Text>{item.title}</S.Text>
        </S.Container>
      </MotiView>
    ))}
  </S.Map>
);

export default CarouselAboutMe;
