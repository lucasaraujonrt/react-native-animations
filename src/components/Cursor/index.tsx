import React from 'react';
import { View, StyleSheet, Dimensions } from 'react-native';
import {
  PanGestureHandler,
  PanGestureHandlerGestureEvent,
} from 'react-native-gesture-handler';
import Animated, {
  Extrapolate,
  interpolate,
  useAnimatedGestureHandler,
  useAnimatedStyle,
  withDecay,
} from 'react-native-reanimated';

import type { Path } from '@mobile/animation/helpers/SVG';
import type { DataPoint } from '@mobile/components/Label';
import theme from '@mobile/theme';

const { width } = Dimensions.get('window');

const CURSOR = 100;
const styles = StyleSheet.create({
  cursorContainer: {
    width: CURSOR,
    height: CURSOR,
    justifyContent: 'center',
    alignItems: 'center',
  },
  cursor: {
    width: 30,
    height: 30,
    borderRadius: 15,
    borderColor: theme.colors.homePurple,
    borderWidth: 4,
    backgroundColor: 'white',
  },
});

interface CursorProps {
  path: Path;
  length: Animated.SharedValue<number>;
  point: Animated.SharedValue<DataPoint>;
}

const Cursor = ({ path, length, point }: CursorProps) => {
  const onGestureEvent = useAnimatedGestureHandler<
    PanGestureHandlerGestureEvent,
    {
      offsetX: number;
      offsetY: number;
    }
  >({
    onStart: (event, ctx) => {
      ctx.offsetX = interpolate(
        length.value,
        [0, path.length],
        [0, width],
        Extrapolate.CLAMP
      );
    },
    onActive: (event, ctx) => {
      length.value = interpolate(
        ctx.offsetX + event.translationX,
        [0, width],
        [0, path.length],
        Extrapolate.CLAMP
      );
    },

    onEnd: ({ velocityX: velocity }) => {
      length.value = withDecay({ velocity, clamp: [0, path.length] });
    },
  });

  const style = useAnimatedStyle(() => {
    const translateX = point.value?.coord?.x - CURSOR / 2;
    const translateY = point.value?.coord?.y - CURSOR / 2;

    return {
      transform: [{ translateX }, { translateY }],
    };
  });

  return (
    <View style={StyleSheet.absoluteFill}>
      <PanGestureHandler {...{ onGestureEvent }}>
        <Animated.View style={[styles.cursorContainer, style]}>
          <View style={styles.cursor} />
        </Animated.View>
      </PanGestureHandler>
    </View>
  );
};

export default Cursor;
