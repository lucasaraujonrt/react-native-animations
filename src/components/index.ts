export { default as Button } from './Button/Button';
export { default as Text } from './Text/Text';
export { default as Input } from './Input';
export { default as Background } from './Background';
export { default as TouchableOpacity } from './TouchableOpacity';
export { default as Row } from './Grid/Row/Row';
export { default as ImageBackground } from './Grid/ImageBackground';
export { default as Col } from './Grid/Col/Col';
