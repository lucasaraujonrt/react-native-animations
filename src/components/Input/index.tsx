import React, { useState } from 'react';
import { TextInputProps } from 'react-native';
import { Text, TouchableOpacity } from '@mobile/components';
import { Container, Input as StyledInput } from './styles';

interface IInputProps extends TextInputProps {
  title: string;
  password?: boolean;
}

const Input = (props: IInputProps) => {
  const [focus, setFocus] = useState(false);
  const [showPassword, setShowPassword] = useState(false);

  const handleFocus = () => setFocus(true);
  const handleBlur = () => setFocus(false);
  const handleShowPassword = () => setShowPassword(false);

  return (
    <Container>
      <Text text={props.title} />
      <StyledInput
        {...props}
        showBorder={focus}
        onFocus={handleFocus}
        onBlur={handleBlur}
        secureTextEntry={props.password && showPassword}
      />
      {showPassword && (
        <TouchableOpacity onPress={handleShowPassword}>
          {/* <Icon
            name={showPassword ? 'eye' : 'eye-with-line'}
            color="#000"
            size={16}
          /> */}
        </TouchableOpacity>
      )}
    </Container>
  );
};

export default Input;
