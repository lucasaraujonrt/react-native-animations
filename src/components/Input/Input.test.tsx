import React from 'react';
import { renderWithTheme } from '@mobile/test-utils';
import { Input } from '@mobile/components';

describe('Input Snapshot', () => {
  const component = renderWithTheme(<Input title="testing" />);

  it('should be render Input component', () => {
    expect(component).toMatchSnapshot();
  });
});
