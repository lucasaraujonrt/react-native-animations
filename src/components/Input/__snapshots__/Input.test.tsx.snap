// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Snapshot should be render Input component 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 34.285714285714285,
        },
      ]
    }
  >
    testing
  </Text>
  <TextInput
    allowFontScaling={true}
    onBlur={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    showBorder={false}
    style={
      Array [
        Object {
          "borderColor": "transparent",
        },
      ]
    }
    title="testing"
    underlineColorAndroid="transparent"
  />
</View>
`;
